frontend/src/components/problem/Selector.tsx

```diff
@@ -198,3 +227,47 @@ export function TagSelector(props: TagSelectorProps) {
     </Content>
   );
 }
+
+export function SpectatorFilter(props: SpectatorSelectorProps) {
+  const { spectators } = props;
+
+  const [showSpectators, setShowSpectators] = useState(false);
```

Comment:
I think this makes sense - in the future if there is a chatbox of some sort, I could imagine it being placed there. Not sure about where else to put it (if anywhere) right now; I would do the player card list at the top, but it's kind of difficult to put the spectators there in a clearly different format.

I think spectators may still need to enter nicknames because it'd require a fairly big redesign otherwise, but I can try changing the current component to simply text and see how that looks. (I could also change it to a dropdown, open to thoughts here.)

Reply:
Here is the screenshot (below) where I replaced the spectator filter simply with text. I structured it as "Spectators (#)" instead of "# Spectators" because the format looked a bit better.

Thoughts @alankbi? Should I make a dropdown or keep this design?

**New Spectator Text on Game Page**
<img width="1279" alt="New Spectator Text on Game Page" src="https://user-images.githubusercontent.com/7987279/120540891-0b7e2e80-c39e-11eb-9faa-017694790332.png">

Reply:
I think this looks great, assuming your cool with having a minimal approach for now - tho one small nit would be to add slightly more space between the room id and this text

Reply:
Ok great! Yeah I'll add a bit more margin, sounds good 👍